### PR TITLE
GH-1185 AmChart Date Sorting Fix

### PR DESCRIPTION
**Pull request by mlzummo**
@GordonSmith @dtsnell4 @jbrundage Please Review

Fixes GH-1185

Signed-off-by: Mathew Zummo <mzummo@gmail.com>

### CODE REVIEW

**Comment by mlzummo** [comment]
need to check what other charts XY? i might need to apply this fix to?

File: src/amchart/CommonSerial.js [hunk 1]
```diff
@@ -228,30 +247,40 @@
...
+                this._chart.categoryAxis.minPeriod = this.axisMinPeriod() ? this.axisMinPeriod() : undefined;
                 this._chart.categoryAxis.logarithmic = false;
+
+                if (this.xAxisTickFormat()) {
+                    this.dataFormatter = d3.time.format(this.xAxisTickFormat());
+                } else if (this.xAxisTypeTimePattern()) {
+                    this.dataFormatter = d3.time.format(this.xAxisTypeTimePattern());
+                } else {
+                    this.dataFormatter =  function(v) { return v; };
+                }
```
Comment (GordonSmith) [inline src/amchart/CommonSerial.js]: I wouldn't mix my "data pattern parser" (string->date) and my display formatter (date->string) up like this.

**Comment by mlzummo** [comment]
@GordonSmith i did this becuase in order to parse dates in a d3 way and not amcharts i need to convert to a full date object via d3 which amcharts will automatically know what to do with ... and im trying to make it not display a full DT string if there is no tick format supplied ... if you would still like it removed let me know

**Comment by GordonSmith** [comment]
If your happy it works (my comment was based on not really understanding what was going on), its ok.

**Pull request merged**